lib/src/building_explorer/building_floor_level_selector.dart

```diff
@@ -0,0 +1,118 @@
+//
+// Copyright 2025 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+part of '../../arcgis_maps_toolkit.dart';
+
+class _BuildingFloorLevelSelector extends StatefulWidget {
+  const _BuildingFloorLevelSelector({required this.buildingSceneLayer});
+
+  final BuildingSceneLayer buildingSceneLayer;
+
+  @override
+  State<StatefulWidget> createState() => _BuildingFloorLevelSelectorState();
+}
+
+// Widget to list and select building floor.
+class _BuildingFloorLevelSelectorState
+    extends State<_BuildingFloorLevelSelector> {
+  // The currently selected floor.
+  var _selectedFloor = 'All';
+
+  // A listing of all floors in the building scene layer.
+  var _floorList = <String>[];
+
+  @override
+  void initState() {
+    super.initState();
+
+    _initFloorList();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final options = ['All', ..._floorList];
+    return Padding(
+      padding: const EdgeInsets.fromLTRB(15, 0, 20, 0),
+      child: Row(
+        // mainAxisAlignment: MainAxisAlignment.spaceAround,
+        children: [
+          Text('Select Level:', style: Theme.of(context).textTheme.bodyLarge),
+          const Spacer(),
+          DropdownButton<String>(
+            value: _selectedFloor,
+            items: options
+                .map(
+                  (value) => DropdownMenuItem(value: value, child: Text(value)),
+                )
+                .toList(),
+            onChanged: onFloorChanged,
+          ),
+        ],
+      ),
+    );
+  }
+
+  Future<void> _initFloorList() async {
+    // Get the floor listing from the statistics.
+    final statistics = await widget.buildingSceneLayer.fetchStatistics();
+    if (statistics['BldgLevel'] != null) {
+      final floorList = <String>[];
+      floorList.addAll(statistics['BldgLevel']!.mostFrequentValues);
```

Comment:
As an aside, this strikes me as a rather unfortunate way to get the list of levels. I never would have guessed to do it this way, if all I had to go on was the API documentation.